File: GithubBrowserSample/app/src/test-common/java/com/android/example/github/util/PagedListUtil.java
package com.android.example.github.util;

import android.arch.core.executor.AppToolkitTaskExecutor;
import android.arch.paging.PagedList;
import android.arch.paging.TiledDataSource;

import java.util.List;

/**
 * helper methods to create static paged lists
 */
public class PagedListUtil {
    public static <T> PagedList<T> from(List<T> source) {
        return new PagedList.Builder<Integer, T>()
                .setBackgroundThreadExecutor(AppToolkitTaskExecutor.getIOThreadExecutor())
                .setMainThreadExecutor(AppToolkitTaskExecutor.getMainThreadExecutor())
                .setDataSource(new TiledDataSource<T>() {
                    @Override
                    public int loadCount() {
                        return source.size();
                    }

                    @Override
                    public List<T> loadRange(int startPosition, int count) {
                        int end = Math.min(source.size(), startPosition + count);
                        return source.subList(startPosition, end);
                    }
                })
                .setConfig(new PagedList.Config.Builder()
                    .setEnablePlaceholders(true)
                    .setInitialLoadSizeHint(Math.max(source.size(), 1))
                    .setPageSize(Math.max(source.size(), 1)).build()).build();

    }
}

File: GithubBrowserSampleKotlin/app/src/test-common/java/com/android/example/github_kotlin/util/PagedListUtil.kt
package com.android.example.github_kotlin.util

import android.arch.core.executor.AppToolkitTaskExecutor
import android.arch.paging.PagedList
import android.arch.paging.TiledDataSource

/**
 * helper methods to create static paged lists
 */
object PagedListUtil {
    fun <T> from(source: List<T>): PagedList<T> {
        return PagedList.Builder<Int, T>()
                .setBackgroundThreadExecutor(AppToolkitTaskExecutor.getIOThreadExecutor())
                .setMainThreadExecutor(AppToolkitTaskExecutor.getMainThreadExecutor())
                .setDataSource(object : TiledDataSource<T>() {

                    override fun countItems(): Int {
                        return source.size
                    }

                    override fun loadRange(startPosition: Int, count: Int): List<T> {
                        val end = Math.min(source.size, startPosition + count)
                        return source.subList(startPosition, end)
                    }
                })
                .setConfig(PagedList.Config.Builder()
                                   .setEnablePlaceholders(true)
                                   .setInitialLoadSizeHint(Math.max(source.size, 1))
                                   .setPageSize(Math.max(source.size, 1)).build()).build()

    }
}
